scripts/generateData/index.js

```diff
@@ -0,0 +1,39 @@
+const chalk = require('chalk');
+const get = require('lodash/get');
+
+/**
+ * Generates tree structure for tree viz
+ * Two pass to account for random order
+ * Generates objects in first pass
+ * Maps objects in second pass
+ */
+function generateTreeData(data) {
```

Comment:
just curious, how is it generating a tree without recursion?

Reply:
@hlminh2000 
`mapping` holds node objects. 
The `children` property of these objects is manipulated in 2nd reduce.
These object references are passed into `tree`

for example:
-- sample_reg
--- schema
---- donor

would look something like
- `sample_reg` has no parent, so add to root of `tree` (line: 31)
- `specimen` has parent of `sample_reg`, so push to `sample_reg.children` object in mapping, which is referenced in `tree`
- `donor` has parent of `specimen`, pushes to `specimen` mapping, which is already referenced in `tree` under `sample_reg`

Have a recursive version in mind? :)

Reply:
nope! this is perfect :) would actually avoid recursion myself:)